src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,199 @@
+        aboutToCompleteBothDates ? selectedDay : daysInitialState
+      );
+
+      if (aboutToCompleteBothDates) {
+        dayPickerRef.current?.hideDayPicker();
+      }
+
+      setSelectedDay(range);
+      setSelectedOption('custom');
+    },
+    [selectedDay, setSelectedDay, setSelectedOption, dayPickerRef.current]
+  );
+
+  const handleDayClick = useCallback(
+    day => {
+      setSelectedDay({ from: day, to: day });
+      setSelectedOption('custom');
+      dayPickerRef.current?.hideDayPicker();
+    },
+    [setSelectedDay, setSelectedOption, dayPickerRef.current]
+  );
+
+  const handleSelectedOptions = (option: string) => {
+    const foundOption = extraOptions.find(optionItem => optionItem.value === option);
+    dayPickerRef.current?.hideDayPicker();
+
+    if (foundOption) {
+      setSelectedDay(
+        Array.isArray(foundOption.dates)
+          ? { from: foundOption.dates[0], to: foundOption.dates[1] }
+          : { from: foundOption.dates, to: foundOption.dates }
+      );
+    }
+
+    setSelectedOption(option);
+  };
+
+  const modifiers = { start: selectedDay.from, end: selectedDay.to };
+  const getDateFormatted = useCallback(
+    (date: Date | undefined) => (date ? dayjs(date).format('MM/DD/YYYY') : ''),
+    []
+  );
+
+  return (
+    <div css={datePickerStyles({ isRangePicker })(theme)}>
+      <DayPickerInput
+        ref={dayPickerRef}
+        overlayComponent={(props: DayPickerInputProps) => (
+          <OverlayComponent
+            selectedOption={selectedOption}
+            setSelectedOption={handleSelectedOptions}
+            extraOptions={extraOptions}
+            isRangePicker={isRangePicker}
+            {...props}
+          />
+        )}
+        dayPickerProps={{
+          onDayClick: isRangePicker ? handleDayRangeClick : handleDayClick,
+          selectedDays: selectedDay as RangeModifier,
+          modifiers: isRangePicker ? modifiers : undefined,
+          firstDayOfWeek: 1,
+          numberOfMonths: isRangePicker ? 2 : 1,
+          disabledDays: disableFutureDates
+            ? [
+                {
+                  after: new Date(),
+                },
+              ]
+            : undefined,
+        }}
+        component={(props: DayPickerInput) =>
+          isRangePicker ? (
+            <div
+              css={[
+                wrapperStyle({})(theme),
+                flex,
+                {
+                  width: 275,
+                },
+              ]}
+            >
+              <TextField
+                label="Date (Start)"
+                lean={true}
+                {...props}
+                value={getDateFormatted(selectedDay.from)}
+              />
+              <TextField
+                rightIcon={<Icon name={'calendarEmpty'} color={'secondary'} />}
+                label={`Date (End)`}
+                {...props}
+                lean={true}
+                value={getDateFormatted(selectedDay.to)}
+              />
+            </div>
```

Comment:
you can extract this to a new component to make the existing component more testable and clean.